### PR TITLE
Add --dry-run option to version command

### PR DESCRIPTION
**Pull request by estyxx**
# Pull Request Check List

Resolves: #5074

This PR add the `--dry-run` option to the `version` command, so it will not update the `pyproject.toml` file but only print the output on the terminal. 

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

### CODE REVIEW

File: src/poetry/console/commands/version.py [hunk 1]
```diff
@@ -29,7 +29,10 @@ class VersionCommand(Command):
             optional=True,
         )
     ]
-    options = [option("short", "s", "Output the version number only")]
+    options = [
+        option("short", "s", "Output the version number only"),
+        option("dry-run", None, "Outputs the operations but will not execute anything"),
```
Comment (abn) [inline src/poetry/console/commands/version.py:34]: Maybe reword this so it is specific to the command and indicate that we do not write aything to the toml file.

File: tests/console/commands/test_version.py [hunk 2]
```diff
@@ -76,3 +76,12 @@ def test_version_update(tester: CommandTester):
 def test_short_version_update(tester: CommandTester):
     tester.execute("--short 2.0.0")
     assert tester.io.fetch_output() == "2.0.0\n"
+
+
+def test_dry_run(tester: CommandTester):
```
Comment (abn) [inline tests/console/commands/test_version.py:81]: :+1:

**Review by abn — CHANGES_REQUESTED** [review]

File: src/poetry/console/commands/version.py (hunk 1 above)
Comment (estyxx) [reply src/poetry/console/commands/version.py]: Updated

File: docs/cli.md [hunk 3]
```diff
@@ -622,6 +622,7 @@ The table below illustrates the effect of these rules with concrete examples.
 ### Options
 
 * `--short (-s)`: Output the version number only.
+* `--dry-run`: Output the operations but do not execute anything.
```
Comment (abn) [inline docs/cli.md:625]: Missed this one.

File: src/poetry/console/commands/version.py [hunk 4]
```diff
@@ -29,7 +29,14 @@ class VersionCommand(Command):
             optional=True,
         )
     ]
-    options = [option("short", "s", "Output the version number only")]
+    options = [
+        option("short", "s", "Output the version number only"),
+        option(
+            "dry-run",
+            None,
+            "Outputs the version without updating the pyproject.toml file",
```
Comment (abn) [inline src/poetry/console/commands/version.py:37]:
```suggestion
            "Do not update pyproject.toml file",
```

This might suffice. Because technically we do not output the version but the version bump log.

File: docs/cli.md (hunk 3 above)
Comment (estyxx) [reply docs/cli.md]: done

**Comment by abn** [comment]
Might want to rebase this on master to get the checks passing.
